onnx/defs/schema.h

```diff
@@ -837,12 +848,16 @@ class OpSchema final {
   const FunctionProto* GetFunction() const;
 
   bool HasContextDependentFunction() const {
-    return functionBuilder_ != nullptr;
+    return opset_version_to_function_builder_.find(since_version_) != opset_version_to_function_builder_.end();
+  }
+
+  bool HasContextDependentFunctionWithOpsetVersion(int opset_version) const {
```

Comment:
I think we can use the same name `HasContextDependentFunction` as an overload (or use  single function with a default value for `opset_version` just like in the functions below.)

Reply:
Do you assume a function will only use the opset the operator is defined into? I'm think8ng about ai.onnx.ml. A couple of operators can be defined as function of existing operators in ai.onnx.

Reply:
> Do you assume a function will only use the opset the operator is defined into? I'm thinking about ai.onnx.ml. A couple of operators can be defined as function of existing operators in ai.onnx.

Good point. I think eventually we should support an interface where `opset_version` parameter is a map from domains to versions (not necessarily in this PR). Xavier's example can be handled right now by treating opset_version as the version of ai.onnx (not the domain the function belongs to). But there are examples in onnxruntime where we would want to mix a custom domain and ai.onnx

Reply:
Currently it assumes ops used to define a function op are from the same opset version of the same opset. With this change, it allows to specific the opset version. Using ops from different opsets is allowed at the ONNX IR level. However I do not see the related support with OpSchema. I agree with Rama that we need a separate PR for this.

Reply:
Do we need `WithOpsetVersion` in the function name? Why not drop it? Overloading will resolve it, right?